XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -346,41 +382,93 @@ protected virtual void Awake()
             }
 
             boundingBox = GetComponent<BoundingBox>();
+
+            body = gameObject.EnsureComponent<Rigidbody>();
+            body.useGravity = false;
+            body.isKinematic = true;
+            body.interpolation = RigidbodyInterpolation.Interpolate;
+            body.collisionDetectionMode = CollisionDetectionMode.ContinuousSpeculative;
+            body.constraints = RigidbodyConstraints.FreezeAll;
         }
 
-        protected virtual void Update()
+        private void Update()
         {
-            if (!IsBeingHeld || primaryPointer == null) { return; }
-
-            var pointerPosition = primaryPointer.Result.EndPoint;
-
-            if (!IsPressed)
-            {
-                if (!IsRotating && !IsScalingPossible)
-                {
-                    manipulationTarget.position = grabbedPosition + pointerPosition;
-                }
-            }
-            else
+            for (var keyCode = KeyCode.Alpha0; keyCode < KeyCode.Tilde; keyCode++)
             {
-                if (IsNudgePossible)
+                if (UnityEngine.Input.GetKey(keyCode))
                 {
-                    manipulationTarget.position = grabbedPosition + pointerPosition;
-                    primaryPointer.PointerExtent = updatedExtent;
+                    const float nudgeFactor = 0.5f;
+                    float prevExtent;
+                    float newExtent;
+
+                    const float scaleFactor = 0.1f;
+                    Vector3 newScale;
+
+                    switch (keyCode)
+                    {
+                        case KeyCode.W:
```

Comment:
This was added to help with testing in the editor. It should either be updated to utilize the input actions, or removed.